MetaMask/metamask-extension
# [e2e] Deploy contracts directly seeding Ganache

### Goal
The main goal is to deploy the smart contracts directly on Ganache, instead of using the Dapp Test interface for achieving the same actions.
These smart contracts can be used later within the e2e tests in order to reduce the amount of code needed to be written in order to execute the same action every time for each of the tests (i.e. contract deployment). This would also reduce the time needed for e2e tests execution because waiting for the UI action response will be eliminated.

### Approach
We are going to split this task in 2 different PRs. The first one (this one) will take care of preparing the contracts and the seeder, so it deploys correctly any contract we pass them into Ganache, from the background.

The second part will take care of the refactor of the e2e testcases, in order to use the smart contract deployer, instead of deploying each contract manually, using the Dapp UI.

**First Part in Detail**
- Inside `e2e/seeder` we have a contracts folder, with the different byte code & abi contracts from Test-Dapp. Having separate contracts helps us to be more granular whenever we want to deploy just 1 contract specifically. It can also be expanded by adding more contracts, following the same approach
- `ganache-seeder.js` will deploy a contract to ganache, by contract name

### Testing Steps
- Include smartContract fixture on any testcase with a contract name. I.e.
![image](https://user-images.githubusercontent.com/54408225/180226444-6151e371-8a65-48e3-984b-ac095f9d40e1.png)
- Run test `yarn test:e2e:single test/e2e/tests/test-name.spec.js --browser=chrome`. 
- Check smart contracthas been deployed to ganache by looking at the console.
